server/index.js

```diff
@@ -68,6 +68,114 @@ app.get('/api/products/:productId', (req, res, next) => {
     );
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!(req.session.cartId)) {
+    res.json([]);
+    return;
+  }
+  const sql = `
+  select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+  from "cartItems" as "c"
+  join "products" as "p" using ("productId")
+ where "c"."cartId" = $1
+  `;
+  const cartId = req.session.cartId;
+  const value = [cartId];
+  db.query(sql, value)
+    .then(result => res.status(200).json(result.rows))
+    .catch(err => next(err));
+
+});
+
+app.post('/api/cart/:productId', (req, res, next) => {
+  const productId = parseInt(req.params.productId);
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return res.status(400).json({
+      error: '"productId" must be a positive integer'
+    });
+  }
+  const sql = `
+  select "price"
+    from "products"
+   where "productId" = $1
+  `;
+  const value = [productId];
+  db.query(sql, value)
+    .then(result => {
+      const productPrice = result.rows[0].price;
+      if (!(result.rows[0])) {
+        return next(new ClientError(`Unable to find product productId ${productId}`, 400));
+      }
```

Comment:
Be sure to `throw new ClientError(message, statusCode)` here instead of calling `next()`. We want to skip to the `.catch()`, but just calling `next()` will allow the promise chain to continue.